cmd/vsphere-xcopy-volume-populator/internal/populator/remote_esxcli.go

```diff
@@ -170,11 +186,51 @@ func (p *RemoteEsxcliPopulator) Populate(sourceVMDKFile string, volumeHandle str
 		response += l.Value("message")
 	}
 
-	go func() {
-		// TODO need to process the vmkfstools stderr(probably) and to write the
-		// progress to a file, and then continuously read and report on the channel
-		progress <- 100
-		quit <- nil
-	}()
-	return nil
+	v := vmkfstoolsClone{}
+	err = json.Unmarshal([]byte(response), &v)
+	if err != nil {
+		return
+	}
+
+	for {
+		r, err = p.VSphereClient.RunEsxCommand(context.Background(),
+			host, []string{"vmkfstools", "taskGet", "-i", v.TaskId})
+		if err != nil {
+			quit <- err
+			return
+		}
+		response := ""
+		klog.Info("respose from esxcli ", r)
+		for _, l := range r {
+			response += l.Value("message")
+		}
+		v := vmkfstoolsTask{}
+		err = json.Unmarshal([]byte(response), &v)
+		if err != nil {
+			return
+		}
+
+		// exmple output - Clone: 20% done.
```

Comment:
Think regex here would be a better fit as I see a lot of parsing
`Clone:\ (\d+)%\ done.` and get the first group